### PR TITLE
feat(test): add a new test to mimic cmd arguments

### PR DESCRIPTION
**Pull request by tareknaser360**
This PR adds a new test to replicate this error #145 
The test now fails but will come in handy to solve the issue

### CODE REVIEW

File: tests/src/coffee_integration_tests.rs [hunk 1]
```diff
@@ -32,6 +33,28 @@ pub async fn init_coffee_test() -> anyhow::Result<()> {
...
+        network: Some("testnet".to_string()),
+    };
+    let coffee = CoffeeManager::new(&args).await?;
+
+    let actual_network = coffee.storage_info().config.network;
+    let expected_network = "testnet".to_string();
+
+    assert_eq!(
+        actual_network, expected_network,
+        "Network is wrong. Actual: '{}', Expected: '{}'",
```
Comment (vincenzopalazzo) [inline tests/src/coffee_integration_tests.rs:52]:
```suggestion
        "Network is wrong. Actual: '{actual_network}', Expected: '{expected_network}'",
```

File: tests/src/coffee_integration_tests.rs [hunk 2]
```diff
@@ -32,6 +33,28 @@ pub async fn init_coffee_test() -> anyhow::Result<()> {
...
+        actual_network, expected_network
```
Comment (vincenzopalazzo) [inline tests/src/coffee_integration_tests.rs:53]:
```suggestion
```

File: tests/src/coffee_integration_tests.rs [hunk 3]
```diff
@@ -32,6 +33,28 @@ pub async fn init_coffee_test() -> anyhow::Result<()> {
...
 }
 
+#[tokio::test]
+pub async fn init_coffee_test_cmd() -> anyhow::Result<()> {
+    init();
+
+    let args = MockCoffeeArgs {
+        conf: None,
+        data_dir: None,
+        network: Some("testnet".to_string()),
```
Comment (vincenzopalazzo) [inline tests/src/coffee_integration_tests.rs:43]:
```suggestion
        network: Some("regtest".to_string()),
```

Try this kind of network

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

File: coffee_testing/src/lib.rs [hunk 4]
```diff
@@ -100,6 +101,19 @@ impl CoffeeTesting {
         })
     }
 
+    // init coffee in a tmp directory with arguments.
+    pub async fn tmp_with_args(args: &CoffeeTestingArgs) -> anyhow::Result<Self> {
+        let temp_dir_path = args.data_dir.clone();
+        let dir = Builder::new().prefix(&temp_dir_path).tempdir()?;
```
Comment (vincenzopalazzo) [inline coffee_testing/src/lib.rs:107]: this is always the correct path?
```suggestion
        let dir = Builder::new().prefix(&temp_dir_path).tempdir()?;
```

**Review by vincenzopalazzo — COMMENTED** [review]
Nice! finally we had the reproducer

File: coffee_testing/src/lib.rs (hunk 4 above)
Comment (tareknaser360) [reply coffee_testing/src/lib.rs:107]: we can take the temp path as an argument to avoid this line of code

Comment (vincenzopalazzo) [reply coffee_testing/src/lib.rs:107]: why not put the tmp path directly inside the args?

Comment (tareknaser360) [reply coffee_testing/src/lib.rs:107]: The args expects a string for the data directory.
what do you think of the latest commit?
I added an info debug statement to make sure they are the same

Comment (vincenzopalazzo) [reply coffee_testing/src/lib.rs]: >The args expects a string for the data directory.

but you can convert the tmp directory to a path and then convert the path to a string no? as I did here https://github.com/coffee-tools/coffee/blob/master/coffee_testing/src/lib.rs#L90

File: coffee_testing/src/lib.rs [hunk 5]
```diff
@@ -100,6 +100,21 @@ impl CoffeeTesting {
         })
     }
 
+    // init coffee in a tmp directory with arguments.
+    pub async fn tmp_with_args(args: &CoffeeTestingArgs) -> anyhow::Result<Self> {
+        let tempdir = tempfile::tempdir_in(args.data_dir.clone())?;
+        log::info!("tempdir: {:?}", tempdir);
```
Comment (vincenzopalazzo) [inline coffee_testing/src/lib.rs:106]: Why do not pass the dir as parameter? No one is stopping you do to it

File: tests/src/coffee_integration_tests.rs [hunk 6]
```diff
@@ -32,6 +31,40 @@ pub async fn init_coffee_test() -> anyhow::Result<()> {
...
 }
 
+#[tokio::test]
+pub async fn init_coffee_test_cmd() -> anyhow::Result<()> {
+    init();
+
+    let dir = tempfile::tempdir()?;
+    let args = CoffeeTestingArgs {
+        conf: None,
+        data_dir: dir.path().to_str().unwrap().to_owned(),
```
Comment (vincenzopalazzo) [inline tests/src/coffee_integration_tests.rs:41]:
```rust
let tmp_path = dir.path().to_str().unwrap().to_owned();
```

and then use it around the code

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

**Comment by vincenzopalazzo** [comment]
This should fix your error, and in my machine the tests is passing

```diff
diff --git a/coffee_testing/src/lib.rs b/coffee_testing/src/lib.rs
index e1dcd50..255b988 100644
--- a/coffee_testing/src/lib.rs
+++ b/coffee_testing/src/lib.rs
@@ -71,14 +71,14 @@ impl coffee_core::CoffeeArgs for CoffeeTestingArgs {
 /// we need to perform integration testing for coffee.
 pub struct CoffeeTesting {
     inner: CoffeeManager,
-    root_path: TempDir,
+    root_path: String,
 }
 
 impl Drop for CoffeeTesting {
     fn drop(&mut self) {
         use std::fs;
 
-        fs::remove_dir_all(self.root_path.path()).unwrap();
+        fs::remove_dir_all(self.root_path.to_owned()).unwrap();
     }
 }
 
@@ -96,7 +96,7 @@ impl CoffeeTesting {
             .map_err(|err| anyhow::anyhow!("{err}"))?;
         Ok(Self {
             inner: coffee,
-            root_path: dir,
+            root_path: dir.path().to_str().unwrap().to_owned(),
         })
     }
 
@@ -113,7 +113,7 @@ impl CoffeeTesting {
 
         Ok(Self {
             inner: coffee,
-            root_path: *tempdir,
+            root_path: tempdir.path().to_str().unwrap().to_owned(),
         })
     }

```

So I think this test needs a little bit more stress? maybe you should force to write to the store?

**Comment by vincenzopalazzo** [comment]
This needs a rebase on master

**Comment by vincenzopalazzo** [comment]
Ok Rebased on top of the current master

**Pull request merged**